Git Tests broken in Git 2.35

# Expected Behavior

Git tests continue to work.

# Actual Behavior

```
$ go test ./pkg/git/...
--- FAIL: TestFetch (0.41s)
    --- FAIL: TestFetch/test-good (0.20s)
        git_test.go:231: Fetch() error = signal: aborted, wantErr false
        git_test.go:262: log message: 'Successfully cloned'
             should contains: 'Error running git [checkout -f FETCH_HEAD]: signal: aborted
            BUG: builtin/checkout.c:1102: should be able to skip past 'refs/heads/' in 'refs/remotes/origin/HEAD'!
            '
        git_test.go:216: ["error" git]: Error running git [checkout -f FETCH_HEAD]: signal: aborted
            BUG: builtin/checkout.c:1102: should be able to skip past 'refs/heads/' in 'refs/remotes/origin/HEAD'!
    --- FAIL: TestFetch/test-clone-with-sparse-checkout (0.22s)
        git_test.go:231: Fetch() error = signal: aborted, wantErr false
        git_test.go:262: log message: 'Successfully cloned'
             should contains: 'Error running git [checkout -f FETCH_HEAD]: signal: aborted
            BUG: builtin/checkout.c:1102: should be able to skip past 'refs/heads/' in 'refs/remotes/origin/HEAD'!
            '
        git_test.go:216: ["error" git]: Error running git [checkout -f FETCH_HEAD]: signal: aborted
            BUG: builtin/checkout.c:1102: should be able to skip past 'refs/heads/' in 'refs/remotes/origin/HEAD'!
FAIL
FAIL    github.com/tektoncd/pipeline/pkg/git    0.444s
FAIL
```

# Additional Info

This appears to be introduced in https://github.com/git/git/commit/9081a421a6dbeeda637df8edb0ebf0da11a05b69. This will cause breakage in Git 2.35. For Googlers - this is broken on our workstations now. 😅 

Confirmed with a friend that Git 2.34.1 is working.

--- a/pkg/git/git.go
+++ b/pkg/git/git.go
@@ -119,12 +119,6 @@ func Fetch(logger *zap.SugaredLogger, spec FetchSpec) error {
 		logger.Warnf("Failed to set http.sslVerify in git config: %s", err)
 		return err
 	}
-	if spec.Revision == "" {
-		spec.Revision = "HEAD"
-		if _, err := run(logger, "", "symbolic-ref", spec.Revision, "refs/remotes/origin/HEAD"); err != nil {
-			return err
-		}
-	}
 
 	fetchArgs := []string{"fetch"}
 	if spec.Submodules {

